mssw/MyDate/lib/MyDate.pm

```diff
@@ -0,0 +1,159 @@
+package MyDate;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Carp qw( croak );
+use Data::Dumper;
+use Time::Piece;
+
+=head1 NAME
+
+MyDate - not yours.
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+=head1 SYNOPSIS
+
+gets the date
+
+=head1 SUBROUTINES/METHODS
+
+=head2 new
+
+instantiate all the things!!
+
+=cut
+
+sub new {
+    my $t = localtime;
+    return bless { t => $t }, $_[0];
+}
+
+=head2 AUTOLOAD
+
+for all your autoload-y needs
+
+=cut
+
+sub AUTOLOAD {
+    my %properties = (
+        date  => 'mday',
+        month => 'fullmonth',
+        year  => 'year',
+    );
+
+    our $AUTOLOAD;
+
+    my ($method) = ( $AUTOLOAD =~ /::(\w+)$/ );
+    if ( grep $method eq $_, keys %properties ) {
```

Comment:
It'd be easier to just do `exists $properties{$method}`, avoiding `O(N)` lookup is what hashtables are for.